tests/TestCase/ORM/MarshallerTest.php

```diff
@@ -1745,4 +1745,27 @@ public function testMergeWithCreate()
         $this->assertNotEmpty($result->errors('author_id'));
         $this->assertNotEmpty($result->errors('thing'));
     }
+
+    /**
+     * Test Model.beforeMarshal event.
+     *
+     * @return void
+     */
+    public function testBeforeMarshalEvent()
+    {
+        $data = [
+            'title' => 'My title',
+            'body' => 'My content'
+        ];
+
+        $marshall = new Marshaller($this->articles);
+
+        $this->articles->eventManager()->attach(function ($e, $data) {
+            $data['title'] = 'Modified title';
```

Comment:
I think allowing data modifications means you are stuck with earlier entities being able to modify the data for later ones. My original point was that this behavior should have assertions for it, as it is the kind of thing that could accidentally get broken.